cmd/nexusUpload.go

```diff
@@ -0,0 +1,138 @@
+package cmd
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"hash"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"crypto/md5"
+	"crypto/sha1"
+
+	piperHttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+)
+
+type artifactDescription struct {
+	ID         string `json:"artifactId"`
+	Classifier string `json:"classifier"`
+	Type       string `json:"type"`
+	File       string `json:"file"`
+}
+
+func nexusUpload(config nexusUploadOptions, telemetryData *telemetry.CustomData) error {
+	var artifacts []artifactDescription
+	err := json.Unmarshal([]byte(config.Artifacts), &artifacts)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Failed to convert artifact JSON '", config.Artifacts, "'")
+	}
+
+	client := piperHttp.Client{}
+	clientOptions := piperHttp.ClientOptions{Username: config.User, Password: config.Password, Logger: log.Entry().WithField("package", "github.com/SAP/jenkins-library/pkg/http")}
+	client.SetOptions(clientOptions)
+
+	groupPath := strings.ReplaceAll(config.GroupID, ".", "/")
+
+	for _, artifact := range artifacts {
```

Comment:
I guess it would be better to take this out here for better testability ...

Reply:
Yes, definitely. I didn't write unit tests, yet, since I didn't really know how I need all this to work.

Reply:
if you take the latest version of the step generator you can get a pretty good idea of the general step structure which then should also be rather simple to test.

Reply:
Done.